### PR TITLE
Truncate texts on metadata attributes table

### PR DESCRIPTION
**Pull request by devjoaov**
[linear issue](https://linear.app/bleu-llc/issue/BAL-186/truncate-texts-on-metadata-attributes-table)

image

image

### CODE REVIEW

File: packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx [hunk 1]
```diff
@@ -36,6 +42,52 @@ const Th = ({ className, children }: CellProps) => (
...
+      {hasOverflow &&
+        (isExpanded ? (
+          <button className="block text-blue-400" onClick={toggleExpanded}>
+            see less
+          </button>
+        ) : (
+          <button className="block text-blue-400" onClick={toggleExpanded}>
+            see more
+          </button>
+        ))}
```
Comment (ribeirojose) [inline packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx:86]:
```suggestion
        (<button className="block text-blue-400" onClick={toggleExpanded}>
            {isExpanded ? "see less" : "see more"}
          </button>)
```

File: packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx [hunk 2]
```diff
@@ -36,6 +42,52 @@ const Th = ({ className, children }: CellProps) => (
...
+  };
+
+  return (
+    <div>
+      <div
+        ref={textRef}
+        className={cn(
+          "break-all",
+          !isExpanded && hasBigword
+            ? "overflow-hidden text-ellipsis line-clamp-3"
```
Comment (ribeirojose) [inline packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx:70]: classnames tip 😛 
```suggestion
          {"overflow-hidden text-ellipsis line-clamp-3": !isExpanded && hasBigword }
```

File: packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx [hunk 3]
```diff
@@ -36,6 +42,52 @@ const Th = ({ className, children }: CellProps) => (
...
 
+function ClampedText({ text }: { text: string }) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [hasOverflow, setHasOverflow] = useState(false);
+  const [hasBigword, setHasBigword] = useState(false);
+  const textRef = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    if (textRef.current) {
+      const numLines = textRef.current?.getBoundingClientRect().height / 16;
```
Comment (ribeirojose) [inline packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx:53]: 🤔 could you add a comment explaining this heuristic a bit more?

File: packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx [hunk 4]
```diff
@@ -36,6 +42,52 @@ const Th = ({ className, children }: CellProps) => (
...
+      setHasOverflow(numLines >= 4);
+    }
+    setHasBigword(text.split(" ").some((word) => word.length > 20));
```
Comment (ribeirojose) [inline packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx:56]: 😄 creative solution! Are we considering 20 characters for any specific reason here? Curious to understand the heuristic a bit more. Also, does this really need a separate state for it? It seems to me you could skip this layout effect altogether and have consts for your overflow/long word states

File: packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx [hunk 5]
```diff
@@ -36,6 +42,52 @@ const Th = ({ className, children }: CellProps) => (
...
+  return (
+    <div>
+      <div
+        ref={textRef}
+        className={cn(
+          "break-all",
+          !isExpanded && hasBigword
+            ? "overflow-hidden text-ellipsis line-clamp-3"
+            : "",
+          !isExpanded && hasOverflow ? "line-clamp-3" : ""
```
Comment (ribeirojose) [inline packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx:72]: classnames tip 😛 
```suggestion
          {
            "overflow-hidden text-ellipsis": !isExpanded && hasBigword,
            "line-clamp-3": hasBigWord || hasOverflow,
          }
```
but couldn't you simply always have `break-all line-clamp-3 text-ellipsis`? It sounds like that would achieve the same result you want -- regardless of any additional logic?

**Review by ribeirojose — DISMISSED** [review]
Looks good! Left a few suggestions to consider

**Review by ribeirojose — DISMISSED** [review]

File: packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx (hunk 4 above)
Comment (devjoaov) [reply packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx:57]: I figured that 20 would be a reasonable size for a large word/hash, I confess that I didn't get too attached to this calculation. 

I tried to do it without any effect but without success

**Pull request merged**